### PR TITLE
feat: Addition of block device information in outputs

### PR DESCRIPTION
**Pull request by schniber**
## Description

This PR adds the block device information in the outputs of the module (for the root, ebs block and also ephemeral devices)

## Motivation and Context

The motivation is to expose the Volume IDs that are created through this module. This would allow users willing to create Cloudwatch Monitoring for these EBS volumes to acheive this programatically. 
This PR fixes: #318 

## Breaking Changes

No

## How Has This Been Tested?
- [X] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [X] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [X] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by schniber** [comment]
Hello @bryantbiggs !

This PR is ready for peer review.
Thanks a lot.

Bests.

**Comment by schniber** [comment]
This comment to avoid this to be marked as stale.
Thanks.

**Comment by schniber** [comment]
This comment to avoid this to be marked as stale.
Thanks.

File: outputs.tf [hunk 1]
```diff
@@ -116,3 +116,21 @@ output "iam_instance_profile_unique" {
   description = "Stable and unique string identifying the IAM instance profile"
   value       = try(aws_iam_instance_profile.this[0].unique_id, null)
 }
+
+################################################################################
+# Block Devices
+################################################################################
+output "root_block_device" {
+  description = "Root block device information"
+  value       = try(aws_instance.this[0].root_block_device[0], null)
```
Comment (bryantbiggs) [inline outputs.tf:125]:
```suggestion
  value       = try(aws_instance.this[0].root_block_device, null)
```

File: outputs.tf [hunk 2]
```diff
@@ -116,3 +116,21 @@ output "iam_instance_profile_unique" {
...
+}
+
+output "ebs_block_devices" {
```
Comment (bryantbiggs) [inline outputs.tf:128]:
```suggestion
output "ebs_block_device" {
```

File: outputs.tf [hunk 3]
```diff
@@ -116,3 +116,21 @@ output "iam_instance_profile_unique" {
...
+  description = "EBS block devices information"
+  value       = try(aws_instance.this[0].ebs_block_device, null)
+}
+
+output "ephemeral_block_devices" {
```
Comment (bryantbiggs) [inline outputs.tf:133]:
```suggestion
output "ephemeral_block_device" {
```

File: examples/complete/outputs.tf [hunk 4]
```diff
@@ -74,6 +74,21 @@ output "ec2_complete_iam_instance_profile_unique" {
...
+  description = "Root block device information"
+  value       = module.ec2_complete.root_block_device
+}
+
+output "ec2_complete_ebs_block_devices" {
+  description = "EBS block devices information"
+  value       = module.ec2_complete.ebs_block_devices
+}
+
+output "ec2_complete_ephemeral_block_devices" {
```
Comment (bryantbiggs) [inline examples/complete/outputs.tf:87]:
```suggestion
output "ec2_complete_ephemeral_block_device" {
```

File: examples/complete/outputs.tf [hunk 5]
```diff
@@ -74,6 +74,21 @@ output "ec2_complete_iam_instance_profile_unique" {
   value       = module.ec2_complete.iam_instance_profile_unique
 }
 
+output "ec2_complete_root_block_device" {
+  description = "Root block device information"
+  value       = module.ec2_complete.root_block_device
+}
+
+output "ec2_complete_ebs_block_devices" {
```
Comment (bryantbiggs) [inline examples/complete/outputs.tf:82]:
```suggestion
output "ec2_complete_ebs_block_device" {
```

File: examples/complete/outputs.tf [hunk 6]
```diff
@@ -74,6 +74,21 @@ output "ec2_complete_iam_instance_profile_unique" {
...
+  description = "EBS block devices information"
+  value       = module.ec2_complete.ebs_block_devices
```
Comment (bryantbiggs) [inline examples/complete/outputs.tf:84]:
```suggestion
  value       = module.ec2_complete.ebs_block_device
```

File: examples/complete/outputs.tf [hunk 7]
```diff
@@ -74,6 +74,21 @@ output "ec2_complete_iam_instance_profile_unique" {
...
+}
+
+output "ec2_complete_ephemeral_block_devices" {
+  description = "Ephemeral block devices information"
+  value       = module.ec2_complete.ephemeral_block_devices
```
Comment (bryantbiggs) [inline examples/complete/outputs.tf:89]:
```suggestion
  value       = module.ec2_complete.ephemeral_block_device
```

File: examples/complete/README.md [hunk 8]
```diff
@@ -66,6 +66,8 @@ No inputs.
 |------|-------------|
 | <a name="output_ec2_complete_arn"></a> [ec2\_complete\_arn](#output\_ec2\_complete\_arn) | The ARN of the instance |
 | <a name="output_ec2_complete_capacity_reservation_specification"></a> [ec2\_complete\_capacity\_reservation\_specification](#output\_ec2\_complete\_capacity\_reservation\_specification) | Capacity reservation specification of the instance |
+| <a name="output_ec2_complete_ebs_block_devices"></a> [ec2\_complete\_ebs\_block\_devices](#output\_ec2\_complete\_ebs\_block\_devices) | EBS block devices information |
+| <a name="output_ec2_complete_ephemeral_block_devices"></a> [ec2\_complete\_ephemeral\_block\_devices](#output\_ec2\_complete\_ephemeral\_block\_devices) | Ephemeral block devices information |
```
Comment (bryantbiggs) [inline examples/complete/README.md:70]:
```suggestion
| <a name="output_ec2_complete_ephemeral_block_device"></a> [ec2\_complete\_ephemeral\_block\_device](#output\_ec2\_complete\_ephemeral\_block\_device) | Ephemeral block device information |
```

File: examples/complete/README.md [hunk 9]
```diff
@@ -66,6 +66,8 @@ No inputs.
 |------|-------------|
 | <a name="output_ec2_complete_arn"></a> [ec2\_complete\_arn](#output\_ec2\_complete\_arn) | The ARN of the instance |
 | <a name="output_ec2_complete_capacity_reservation_specification"></a> [ec2\_complete\_capacity\_reservation\_specification](#output\_ec2\_complete\_capacity\_reservation\_specification) | Capacity reservation specification of the instance |
+| <a name="output_ec2_complete_ebs_block_devices"></a> [ec2\_complete\_ebs\_block\_devices](#output\_ec2\_complete\_ebs\_block\_devices) | EBS block devices information |
```
Comment (bryantbiggs) [inline examples/complete/README.md:69]:
```suggestion
| <a name="output_ec2_complete_ebs_block_device"></a> [ec2\_complete\_ebs\_block\_device](#output\_ec2\_complete\_ebs\_block\_device) | EBS block device information |
```

File: README.md [hunk 10]
```diff
@@ -268,6 +268,8 @@ No modules.
 | <a name="output_ami"></a> [ami](#output\_ami) | AMI ID that was used to create the instance. |
 | <a name="output_arn"></a> [arn](#output\_arn) | The ARN of the instance |
 | <a name="output_capacity_reservation_specification"></a> [capacity\_reservation\_specification](#output\_capacity\_reservation\_specification) | Capacity reservation specification of the instance |
+| <a name="output_ebs_block_devices"></a> [ebs\_block\_devices](#output\_ebs\_block\_devices) | EBS block devices information |
```
Comment (bryantbiggs) [inline README.md:271]:
```suggestion
| <a name="output_ebs_block_device"></a> [ebs\_block\_device](#output\_ebs\_block\_device) | EBS block device information |
```

File: README.md [hunk 11]
```diff
@@ -268,6 +268,8 @@ No modules.
...
+| <a name="output_ephemeral_block_devices"></a> [ephemeral\_block\_devices](#output\_ephemeral\_block\_devices) | Ephemeral block devices information |
```
Comment (bryantbiggs) [inline README.md:272]:
```suggestion
| <a name="output_ephemeral_block_device"></a> [ephemeral\_block\_device](#output\_ephemeral\_block\_device) | Ephemeral block device information |
```

File: examples/complete/outputs.tf [hunk 12]
```diff
@@ -74,6 +74,21 @@ output "ec2_complete_iam_instance_profile_unique" {
   value       = module.ec2_complete.iam_instance_profile_unique
 }
 
+output "ec2_complete_root_block_device" {
+  description = "Root block device information"
+  value       = module.ec2_complete.root_block_device
+}
+
+output "ec2_complete_ebs_block_devices" {
+  description = "EBS block devices information"
```
Comment (bryantbiggs) [inline examples/complete/outputs.tf:83]:
```suggestion
  description = "EBS block device information"
```

File: examples/complete/outputs.tf [hunk 13]
```diff
@@ -74,6 +74,21 @@ output "ec2_complete_iam_instance_profile_unique" {
...
+  value       = module.ec2_complete.ebs_block_devices
+}
+
+output "ec2_complete_ephemeral_block_devices" {
+  description = "Ephemeral block devices information"
```
Comment (bryantbiggs) [inline examples/complete/outputs.tf:88]:
```suggestion
  description = "Ephemeral block device information"
```

File: outputs.tf [hunk 14]
```diff
@@ -116,3 +116,21 @@ output "iam_instance_profile_unique" {
...
+  description = "Ephemeral block devices information"
```
Comment (bryantbiggs) [inline outputs.tf:134]:
```suggestion
  description = "Ephemeral block device information"
```

File: outputs.tf [hunk 15]
```diff
@@ -116,3 +116,21 @@ output "iam_instance_profile_unique" {
...
+################################################################################
+# Block Devices
+################################################################################
+output "root_block_device" {
+  description = "Root block device information"
+  value       = try(aws_instance.this[0].root_block_device[0], null)
+}
+
+output "ebs_block_devices" {
+  description = "EBS block devices information"
```
Comment (bryantbiggs) [inline outputs.tf:129]:
```suggestion
  description = "EBS block device information"
```

**Review by bryantbiggs — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 4.4.0 :tada:

**Pull request merged**